integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/saltpassword/RotateSaltPasswordE2ETest.java

```diff
@@ -0,0 +1,139 @@
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.testcase.e2e.AbstractE2ETest;
+import com.sequenceiq.it.cloudbreak.util.spot.UseSpotInstances;
+import com.sequenceiq.it.cloudbreak.util.ssh.action.SshSaltPasswordActions;
+import com.sequenceiq.sdx.api.model.SdxClusterStatusResponse;
+
+public class RotateSaltPasswordE2ETest extends AbstractE2ETest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RotateSaltPasswordE2ETest.class);
+
+    private static final LocalDate PAST_DATE = LocalDate.now().minusMonths(1);
+
+    private static final AtomicReference<String> SHADOW_LINE_REFERENCE = new AtomicReference<>();
+
+    @Inject
+    private SshSaltPasswordActions sshSaltPasswordActions;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private DistroXTestClient distroXTestClient;
+
+    @Override
+    protected void setupTest(TestContext testContext) {
+        createDefaultUser(testContext);
+        createDefaultCredential(testContext);
+        initializeDefaultBlueprints(testContext);
+        createDefaultEnvironment(testContext);
+        createFreeipa(testContext);
+    }
+
+    @Test(dataProvider = TEST_CONTEXT)
+    @UseSpotInstances
+    @Description(
+            given = "there is a running environment with FreeIPA ",
+            when = "rotating a salt password ",
+            then = "FreeIPA should have a new salt password and password expiry ",
+            and = "the same applies to both Datalake and Datahub"
+    )
+    public void testRotateSaltPassword(TestContext testContext) {
+        testContext
+                .given(FreeIpaTestDto.class)
+                .then((testContext1, testDto, client) -> preSaltPasswordRotation(testDto, getFreeipaIpAddresses(testDto)))
+                .when(freeIpaTestClient.rotateSaltPassword())
+                .awaitForFlow()
+                .await(Status.AVAILABLE)
+                .then((testContext1, testDto, client) -> validateSaltPasswordRotation(testDto, getFreeipaIpAddresses(testDto)))
+                .validate();
+        LOGGER.info("FreeIPA salt password rotation test PASSED");
+        SHADOW_LINE_REFERENCE.set("");
+
+        createDatalakeWithoutDatabase(testContext);
+        testContext.given(SdxInternalTestDto.class)
+                .then((testContext1, testDto, client) -> preSaltPasswordRotation(testDto, getSdxIpAddresses(testDto)))
+                .when(sdxTestClient.rotateSaltPassword())
+                .awaitForFlow()
+                .await(SdxClusterStatusResponse.RUNNING)
+                .then((testContext1, testDto, client) -> validateSaltPasswordRotation(testDto, getSdxIpAddresses(testDto)))
+                .validate();
+        LOGGER.info("SDX salt password rotation test PASSED");
+
+        createDefaultDatahubForExistingDatalake(testContext);
```

Comment:
Here you can apply test case specific Data Hub creation steps, as in case in other test cases, or you can use the general ones: 
```
initiateDefaultDatahubCreation(testContext);
waitForDefaultDatahubCreation(testContext);
```

Reply:
The called method does just that

Reply:
Ok. I can get along with your idea.